pyro/contrib/gp/likelihoods/binary.py

```diff
@@ -40,9 +40,11 @@ def forward(self, f_loc, f_var, y=None):
         """
         # calculates Monte Carlo estimate for E_q(f) [logp(y | f)]
         f = dist.Normal(f_loc, f_var.sqrt())()
-        f_res = self.response_function(f)
-
-        y_dist = dist.Bernoulli(f_res)
+        if self.response_function is torch.sigmoid:
+            y_dist = dist.Bernoulli(logits=f)
+        else:
+            f_res = self.response_function(f)
+            y_dist = dist.Bernoulli(f_res)
         if y is not None:
-            y_dist = y_dist.expand_by(y.shape[:-f_res.dim()]).to_event(y.dim())
+            y_dist = y_dist.expand_by(y.shape[:-f.dim()]).to_event(y.dim())
```

Comment:
Can you confirm that `f.dim()` is never 0? Because if `f.dim()` is ever zero, you'll need the more complex expression `y.shape[:y.dim() - f.dim()]`.

Reply:
Yes. Kernels only allow [inputs to be 1D or 2D](https://github.com/pyro-ppl/pyro/blob/dev/pyro/contrib/gp/kernels/kernel.py#L52), so `f` will be at least 1D, even when we are applying GP on 1 data point.